Vscode : release version 0.2.2 of the vscode extension

Alidra opened the pull request:
This version implements the following changes/fixes :

- code refactoring of the client for maintenability.
- fix the bug that causes the proof navigation to malfunction when the `Goals` panel is closed by the user. Now the panel is recreated whenever needed. If focus is taken away frol the `Goals` panel, focus is given back to it when user starts navigating proofs again.
- fix bug related to navigating sub-goals : navigating next subgoal stops before `{` instead of after it so that next subgoal is correctly shown in the `Goals` panel.
- change navigation with ``navigate until cursor`` : Navigation includes the command if the cursor is whithin its range instead of the line above. 
- first command is no more systematically navigated. If the current command is the first one, navigating the previous command results in no command being navigated.

Note that the second point in the above list requires the last modifications in the Lambdapi server (currently in master) to work properly. However, the modifications do not seem to break anything if used with a previous version of Lambdapi (At least not the last released one from the market)

fblanqui commented:
Hi Abdelghani. Thank you for your PR. Note that CI/build_vscode_extension failed.

fblanqui commented on editors/vscode/CHANGES.md:
```diff
@@ -3,9 +3,12 @@ All notable changes to this project will be documented in this file.
 and this project adheres to [Semantic Versioning](https://semver.org/).
 
-## [Unreleased]
+## [0.2.2]
```
Shouldn't it be 0.1.3 instead?

Alidra commented:
> Hi Abdelghani. Thank you for your PR. Note that CI/build_vscode_extension failed.

Indeed. The secret name is incorrect.

Alidra commented:
@fblanqui. 
I corrected the name of the secret in the workflow file. I also added an additional condition to unsure that the extension is published only when pushing to master branch (and not to feature branches). I added some comments too in the Makefile to explain when the extension will be published.

fblanqui commented on Makefile:
```diff
@@ -182,12 +182,27 @@ JQ = $(shell which jq)
 ifeq ($(EXT), $(shell cat editors/vscode/package.json | jq '.version'))
 	echo "extension already exists. Skip"
 else
+ifeq ($(PAT),)
```
shouldn't it be VSCODE_PAT as mentioned in RELEASE.md ?

Alidra replied:
VSCODE_PAT is the name of the secret in gitHub. PAT is the name of the env variable (as defined in the workflow file line 55)
